### PR TITLE
Fix for Winrm Authentication Failure issue during bootstrap

### PR DESCRIPTION
**Pull request by dheerajd-msys**
Added local user account into the Administrators group
In-Progress

### CODE REVIEW

File: lib/chef/knife/ec2_server_create.rb [hunk 1]
```diff
@@ -927,8 +927,10 @@ def eip_scope
       end
 
       def ssl_config_user_data
-<<-EOH
 
+<<-EOH
+net user /add #{locate_config_value(:winrm_user).delete('.\\')} #{windows_password};
+net localgroup Administrators /add #{locate_config_value(:winrm_user).delete('.\\')};
```
Comment (btm) [inline lib/chef/knife/ec2_server_create.rb]: I suppose there isn't a circumstance where someone will be using this feature to bootstrap a server in a domain on ec2 over ssl, but it does seem like we should only being doing this step if we're given a local user.

File: lib/chef/knife/ec2_server_create.rb [hunk 2]
```diff
@@ -927,8 +927,16 @@ def eip_scope
       end
 
       def ssl_config_user_data
+      user_related_commands = ""
+      temp = locate_config_value(:winrm_user).split("\\")
```
Comment (NimishaS) [inline lib/chef/knife/ec2_server_create.rb]: The variable name `temp` should be changed.

File: lib/chef/knife/ec2_server_create.rb [hunk 3]
```diff
@@ -927,8 +927,16 @@ def eip_scope
       end
 
       def ssl_config_user_data
+      user_related_commands = ""
+      temp = locate_config_value(:winrm_user).split("\\") 
+      if (temp[0] == ".") || (temp.length == 1)
```
Comment (Aliasgar16) [inline lib/chef/knife/ec2_server_create.rb]: this would fail for input like `\ec2user`

File: spec/unit/ec2_server_create_spec.rb [hunk 4]
```diff
@@ -1501,7 +1503,7 @@
       end
     end
 
-    context 'ssl config data already exist in user supplied user_data' do
+    context 'ssl config data already exist in user supplied user_data for domain user' do
```
Comment (Aliasgar16) [inline spec/unit/ec2_server_create_spec.rb]: separate `context` should be their for `domain user` and `local user`

**Comment by btm** [comment]
:+1:

**Pull request merged**